documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -2288,21 +2288,21 @@ The following would be invalid:
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
 
-### <a name="r4006" ></a>R4006 XmsCodeGenerationSettingDeprecated
+### <a name="r4006" ></a>R4006 DeprecatedXmsCodeGenerationSetting
 
 **Category** : ARM Warning
 
-**Applies to** : ARM OpenAPI(swagger) specs
+**Applies to** : ARM and Data plane OpenAPI(swagger) specs
 
-**Output Message** : The x-ms-code-generation-settings extenison was deprecated. Consider remove it from the swagger.  
+**Output Message** : The x-ms-code-generation-setting extension is being deprecated. Please remove it and move settings to readme file for code generation.
 
-**Description** : New autorest (V3) will stop supporting “x-ms-code-generation-settings” inside the Swagger. This was designed before there was Readme.md to hold configuration options, and this node should disappear and be replaced by actual readme descriptions.
+**Description** : x-ms-code-generation-settings is being deprecated. AutoRest (v3) will stop supporting it inside the Swagger. There was Readme.md to hold the same configuration options. Please ensure to remove the parameter from swagger spec and move settings to readme.
```

Comment:
avoid using past tense, or future tense.  simple present tense is more objective.

My proposal is 
x-ms-code-generation-settings” is being deprecated. AutoRest (v3) is using settings in readme file for code generation. Please ensure to remove the parameter from swagger spec and move settings to readme.

Reply:
how about:
x-ms-code-generation-settings is being deprecated. AutoRest (v3) is using settings in readme file for code generation and will stop supporting it inside the swagger file. Please ensure to remove the parameter from swagger spec and move settings to readme.

I think it's necessary to point that autorest will stop spport it